src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {useEffect, useRef} from 'react';
+import Nav from 'rsuite/lib/Nav/index.js';
+import styles from '../styles/sidebar.module.css';
+
+const ITEM_HEIGHT = 42;
+const SIDEBAR_HEIGHT = 300;
+
+let timer;
+
+export default function Sidebar({section, onChange, providers}) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (section.eventKey == null) return;
+
+    clearTimeout(timer);
+
+    timer = setTimeout(() => {
```

Comment:
similar comment regarding global state in components. stateful data should not be stored outside of the component context unless it's a store.